components/datetime/src/date.rs

```diff
@@ -2,25 +2,33 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use displaydoc::Display;
 use icu_locid::Locale;
 use std::convert::TryFrom;
 use std::ops::{Add, Sub};
 use std::str::FromStr;
-use thiserror::Error;
 use tinystr::TinyStr8;
 
-#[derive(Error, Debug)]
+#[derive(Display, Debug)]
 pub enum DateTimeError {
-    #[error(transparent)]
-    Parse(#[from] std::num::ParseIntError),
-    #[error("{field} must be between 0-{max}")]
+    #[displaydoc("{0}")]
+    Parse(std::num::ParseIntError),
+    #[displaydoc("{field} must be between 0-{max}")]
     Overflow { field: &'static str, max: usize },
-    #[error("{field} must be between {min}-0")]
+    #[displaydoc("{field} must be between {min}-0")]
     Underflow { field: &'static str, min: isize },
-    #[error("Failed to parse time-zone offset")]
+    #[displaydoc("Failed to parse time-zone offset")]
     InvalidTimeZoneOffset,
 }
 
+impl std::error::Error for DateTimeError {}
```

Comment:
Issue: when `#[error(transparent)]` or `#[from]` is used, I think `thiserror` properly implements the `source` function of `std::error::Error`.  Should we do that here, too?

Reply:
We're not using `.source()` and it's not super important (this is, like I said, some of the "old error handling" stuff that will eventually be replaced)